### PR TITLE
fix: gwPubSub config and sync endpoint 

### PR DESCRIPTION
**Pull request by koonopek**
#15

### CODE REVIEW

File: .env.defaults [hunk 1]
```diff
@@ -19,8 +19,6 @@ GW_HOST=''
 GW_USERNAME=default
 GW_PASSWORD=''
 GW_TLS=true
-GW_ENABLE_OFFLINE_QUEUE=true
-GW_LAZY_CONNECT=true
```
Comment (koonopek) [inline .env.defaults:23]: This envs are not used anywhere, so I removed them

File: README.md [hunk 2]
```diff
@@ -220,13 +220,19 @@ using environment variables or .env file.
...
+| PUBSUB_TYPE           | true                     | Transport for node pub/sub. The node gets information about new blocks with it. Can be `streamr` or `redis`. |
+| STREAMR_STREAM_ID     | if PUBSUB_TYPE=`streamr` | Required when PUBSUB_TYPE is streamr. Streamr stream id                                                      |
+| GW_PORT               | if PUBSUB_TYPE='redis'   | Port to Redis pubsub instance                                                                                |
+| GW_HOST               | if PUBSUB_TYPE='redis'   | Host to Redis pubsub instance                                                                                |
+| GW_USERNAME           | if PUBSUB_TYPE='redis'   | Username for authenticating to Redis pubsub instance                                                         |
+| GW_PASSWORD           | if PUBSUB_TYPE='redis'   | Password for authenticating to Redis pubsub instance                                                         |
+| GW_TLS                | if PUBSUB_TYPE='redis'   | TLS enabled/disabled for Redis pubsub instance                                                               |
```
Comment (koonopek) [inline README.md:234]: Add info about gw configuration

File: README.md [hunk 3]
```diff
@@ -246,7 +252,8 @@ You can check out the default values in the .env.defaults file.
 2. You need to have Arweave wallet. You can generate it using:
     1. [arweave wallet](https://arweave.app/add). You can download the wallet using the download button in
        the [wallet setting](https://arweave.app/settings).
-    2. Generate a new wallet using script: `yarn generate-arweave-wallet`
+    2. Generate a new wallet using script: `mkdir .secrets && yarn generate-arweave-wallet`
+    3. Script will generate arweave keys, you should past JSON string in `NODE_JWK_KEY` env variable in `.env` file
```
Comment (koonopek) [inline README.md:256]: This was missing from my perspective

File: src/config.js [hunk 4]
```diff
@@ -56,7 +54,7 @@ const config = {
   nodeJwk,
   evaluationOptions,
   warpSdkConfig,
-  nodeManifest: (async () => await getNodeManifest())(),
```
Comment (koonopek) [inline src/config.js:59]: ?

File: src/routes/scheduleSync.js [hunk 5]
```diff
@@ -16,7 +16,7 @@ module.exports = async (ctx) => {
       throw new Error(`Chill out and wait ${chillOutTimeSeconds}s`);
     }
     const test = ctx.query.test !== 'false';
-    const connectionOptions = readGwPubSubConfig();
+    const connectionOptions = config.gwPubSubConfig;
```
Comment (koonopek) [inline src/routes/scheduleSync.js:19]: Fix, howver needing testing waiting for: https://github.com/warp-contracts/warp-dre-node/issues/15#issuecomment-1382955269

File: README.md [hunk 6]
```diff
@@ -212,6 +212,25 @@ and ignored.
 3. `/cached` - returns a list of all cached contracts
 4. `/blacklist` - returns all contracts that failed at least once. If contract failed 3 times - it is blacklisted.
 5. `/errors` - returns all errors for all contracts
+6. `/sync` - schedule force synchronization of a given contract. This endpoint can't be called more frequent then every 10 seconds.
```
Comment (koonopek) [inline README.md:215]: Tested

File: .env.defaults (hunk 1 above)
Comment (ppedziwiatr) [reply .env.defaults:23]: are you sure? https://github.com/warp-contracts/warp-dre-node/blob/main/src/config.js#L36-L37

Comment (koonopek) [reply .env.defaults:23]: Yep cause, `config.enableOfflineQueue` is not used later and same parameter exists for `bullmqConfig` so i think it is some kind of "chochlik" here :)

Comment (ppedziwiatr) [reply .env.defaults:23]: Maybe I'm missing sth, but what about https://github.com/warp-contracts/warp-dre-node/blob/main/src/listener.js#L281 ? The configuration object is passed into the Redis instance.

This configuration is used when creating the Redis client for subscribing for notifications from the Warp Gateway (about new interactions and new contracts).

The `bullmqConfig` in related to the internal Redis configuration for the bullmq.

Comment (koonopek) [reply .env.defaults]: Ohh totally true, my bad, thanks for catching this! 
Fixed

**Review by ppedziwiatr — APPROVED** [review]

**Pull request merged**